### ISSUE TITLE
[tests] use `.secret` or `.arg` for build args rather than using branching logic

### ISSUE THREAD

**Issue by nelsam** [earthly/earthly#3020]
GHA yaml files use `if: <condition>` as a field on a step to determine if the step runs. We're using this pretty liberally to run one step if the branch is trusted, or a different step if it's not. Unfortunately, that makes our GHA workflows extremely hard to follow.

We should be able to create the same branching logic _once_, when setting up our `.arg` and/or `.secret` file(s), and drop the rest of the branching logic. This will simplify our workflows, making them both easier  to read and easier to maintain long-term.

**Comment by alexcb** [comment]
done in d875bf0883cb61b717f90e0fd613dbf84b7ef3c7
